### ISSUE TITLE
Rule 932270 False Positives for Tilde + Number

### ISSUE THREAD

**Issue by ssigwart** [coreruleset/coreruleset#3844]
### Description

In version 4.6.0, rule 932270 seems a little too aggressive on the tilde + number matches. For example, "~20 minutes" triggers a false positive. I wonder if `https://github.com/coreruleset/coreruleset/blob/main/regex-assembly/932270.ra` can be updated to check for word boundaries or something.

### How to reproduce the misbehavior (-> curl call)

- Set up a server with CRS 4.6.0.
- Add a GET parameter with a value of `~20 minutes` (`~20+minutes` URL encoded).

### Your Environment

* CRS version (e.g., v3.3.4): 4.6.0
* Paranoia level setting (e.g. PL1) : 1
* ModSecurity version (e.g., 2.9.6): 3.0.12
* Web Server and version or cloud provider / CDN (e.g., Apache httpd 2.4.54): Nginx
* Operating System and version: Linux

### Confirmation

[X] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by RedXanadu** [comment]
Issue confirmed.

Curl call:

```
curl -v -o/dev/null 'localhost/?foo=it+takes+~20+minutes'
```

Log:

``
[Thu Oct 03 16:42:05.587601 2024] [security2:error] [pid 59:tid 131609653151424] [client 172.18.0.1:38586] [client 172.18.0.1] ModSecurity: Warning. Pattern match "~(?:[\\\\+\\\\-](?:$|[\\\\s\\\\x0b0-9]+)|[0-9]+)" at ARGS:foo. [file "/etc/modsecurity.d/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"] [line "398"] [id "932270"] [msg "Remote Command Execution: Unix Shell Expression Found"] [data "Matched Data: ~20 found within ARGS:foo: it takes ~20 minutes"] [severity "CRITICAL"] [ver "OWASP_CRS/4.8.0-dev"] [tag "modsecurity"] [tag "application-multi"] [tag "language-shell"] [tag "platform-unix"] [tag "attack-rce"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/152/248/88"] [hostname "localhost"] [uri "/"] [unique_id "Zv7JXUTZVnBcPQylYzYr_AAAAFg"]
``

**Comment by RedXanadu** [comment]
I can see the logic of what rule 932270 is trying to achieve.

The other things it detects, like `~+2` and `~-`, are very unlikely to ever appear in natural language, but something like `~20` is very likely to do so.

Moving the pattern `~\d+` to a higher paranoia level is the obvious solution.

**Comment by cello86** [comment]
Do you have some feedback about this problem?

**Comment by cello86** [comment]
@airween the test was fine and I commented the PR with my tests.

**Comment by Xhoenix** [comment]
Any ideas which Paranoia Level would be better to the `~\d+`? I think PL2 is not suitable, and we should move it to PL2, any suggestions?
